proposal: CPC-CHARTER to Stage 3 (#166)
openjs-foundation/bootstrap: proposals/stage-3/CPC_CHARTER/CPC-CHARTER.md

# Cross Project Council (CPC) Charter

## Section 1. Guiding Principle.

The OpenJS Foundation will operate transparently, openly,

> collaboratively, and ethically. Project proposals, timelines, and status

Reviewer: ```suggestion
collaboratively, and ethically. Project proposals, timelines, and statuses
```

> must not merely be open, but also easily visible to outsiders.

Reviewer: Interpreted in a straightforward way, this sentence is meaningless. The phrase "easily visible to outsiders" is supposed to be in contrast to "open". Without further explanation, it's not clear what the distinction is supposed to be.

Reviewer: Easy possible solutions would be to remove the sentence or to simply state: "Project proposals, timelines, and statuses must be open." Or: "Project proposals, timelines, and statuses must be easily visible." Or "easily visible to outsiders." Or "easily visible to anyone." Lots of options.

Author: I think we should expand on this in a follow up PR

Reviewer 2: I believe this text came from an earlier doc (not sure if TSC governance, something from the JSF or possibly Kubernetes) so I agree we should expand in a follow on PR.

Author: Not going to include this in these changes. Lets do a follow up

## Section 2. Evolution of OpenJS Foundation Governance.

Most large, complex open source communities have both a business and a

> technical governance model. OpenJS Foundation’s technical leadership

Reviewer: This first sentence of this paragraph should be omitted IMO.

Author: I think we should change that in a follow up PR

is the Cross Project Council (“CPC”). OpenJS Foundation’s business
leadership is the Board of Directors (the “Board”).

This Cross Project Council Charter reflects a carefully
constructed, balanced role for the CPC and the Board in the governance of
OpenJS Foundation. The charter amendment process is for the
OpenJS Foundation to propose changes using simple majority of the
full OpenJS Foundation, the proposed changes being subject to review
and approval by the Board. The Board may additionally make amendments to
the OpenJS Foundation charter at any time, though the Board will not interfere
with day-to-day discussions, votes or meetings
of the CPC.

## Section 3. Board’s Role in Setting OpenJS Foundation’s Strategic Direction. 

> The Board will set the overall CPC Policy. The policy will describe the

Reviewer: ```suggestion
The Board will set the CPC Policy. The policy will describe the
```

overarching scope of the OpenJS Foundation initiative, OpenJS Foundation 
Foundation’s technical vision and direction and project release
expectations in the form of expected cadence and intent. The Board will
use the CPC as a delegate body for governing technical implementation,
individual project scope and direction while they remain within the scope
and direction of the policies as described in the CPC Policy document and
approved by the Board.

> The Board sets the overall CPC policy through [Foundation mission and vision

Reviewer: ```suggestion
The Board sets the CPC policy through [Foundation mission and vision
```

statements][] as well as within the [Foundation bylaws][]. The policy describes
the overarching scope of the OpenJS Foundation initiatives , technical vision,
and direction. The Board uses the CPC as a delegate body for governing
high-level technical policy and procedures while remaining within the scope and
direction of the policies set by the Board.

## Section 4. Establishment of the CPC.

The CPC is an inclusive group with three types of participants:

* Observers
* Regular members
* Voting members

### Observers

Observers are free to attend meetings and participate in the work of the

> CPC as well as the consensus seeking process.

Reviewer: This grants Observers the ability to block things that have consensus among CPC members. If that's intentional, cool. But if not, uh, you'll want to change that.

Author: That is intentional.

### Regular members

Regular members have made a commitment to be involved in an ongoing manner
and take on roles and responsibilities as outlined below.

> Regular member is implied when memership type is not

Reviewer: ```suggestion
Regular member is implied when membership type is not
```

specified. Anyone who has been a member of one of the projects under the 
OpenJS Foundation for at least three months may request to
become a regular member by opening a PR to add themselves to the list of
regular members. Regular members remain for as long as they are active
within the work of the CPC. Regular members who have not been active
in GitHub, participated in meetings, or other work of the CPC for 3 months may be
removed from the list of Regular members. In addition, a Regular CPC member
can be removed by voluntary resignation, or by a standard CPC motion.

### Voting members

Voting members are selected as follows:

* Each Impact project may nominate up to two members through a process
of their choosing. Once nominated the member must be ratified by the
CPC Voting members before becoming a Voting member.

> * up to two Voting members may be nominated by the non Impact projects based

Reviewer: ```suggestion
* up to two Voting members may be nominated by the non-Impact projects based
```

as any other agenda item.

No more than one-fourth of the Voting CPC members may be affiliated with the
same employer. Removal or resignation of a Voting CPC member, or a change of
employment by a Voting CPC member could cause more than
one-fourth of the Voting CPC membership to be affiliated with the same employer.
Such a situation must be immediately remedied by the resignation or
removal of one or more voting CPC members.

The public portion of the CPC discussions and meetings will be open to all
observers and members.

The CPC shall meet regularly using tools that enable participation by the
community. The meeting shall be directed by the CPC Chairperson.
Responsibility for directing individual meetings may be
delegated by the CPC Chairperson to any other CPC member. The CPC chairperson
shall be one of the Voting members, and will be selected by the CPC members
through consensus or if necessary a vote as described in the section titled "Voting".
Minutes or an appropriate recording shall be taken and made available to
the community through accessible public postings.

Voting members have the roles and responsibilities as outlined below.

> Section 5. Responsibilities and Expectations of the CPC

Reviewer 3: This is missing some markdown headers

Subject to such policies as may be set by the Board, the CPC is responsible for
ensuring: 

   1. When needed, members of the CPC will be expected to create subcommittees
      or working groups. The CPC will delegate responsibilities and empower
      these groups to make decisions. Any of the responsibilities listed below
      not identified as being responsibilities of the Voting members may be delegated.

>       For the remanining responsibilities, day-to-day work, investigation, and building

Reviewer: ```suggestion
      For the remaining responsibilities, day-to-day work, investigation, and building
```

>       recommentations may be delegated, however, the final responsibility will

Reviewer: ```suggestion
      recommendations may be delegated, however, the final responsibility will
```

      review/approval the voting members will ensure that the updated charter is compatible
      with the mission of the OpenJS Foundation and reach consensus with the Board on more
      substantial changes before final approval by CPC. 
   1. Voting members are responsible for approving funds for budgets delegated 
      to the project.
   1. Following and be up to date on Board/OpenJS Foundation initiatives and communicate them to the projects.
   1. Defining common practices to be implemented across OpenJS Foundation projects, if any.
   1. Mediating technical conflicts between OpenJS Foundation Projects when attempts to resolve
      those conflicts within the Project were unsuccessful. 
   1. Managing the process for creating a yearly budget and approving a yearly budget to be
      presented to the board.
   1. The CPC serves as the OpenJS Foundation's primary technical
      liaison body with external open source projects, consortiums and groups and is
      also responsible for ensuring that the OpenJS Foundation has appropriate technical
      participation in standards bodies by finding an encouraging members from

>       individual projects to particiapte in these bodies.

Reviewer: ```suggestion
      individual projects to participate in these bodies.
```

   1. Managing the Individual membership program.
   1. Managing programs to improve inclusivity and diversity.
   1. Guiding projects into appropriate Foundation paths.
   1. Shared resources, tooling and funding for projects and Commons.js (Events
      Collaborator Summit, Travel fund).
   1. Supporting projects in their enforcement of Code of Conduct
      violations and escalation.
   1. Developing a framework for community end-user feedback.
   1. Managing programs to offer mentorship to external individuals.
   1. Managing developer outreach programs.
   1. Supporting project advocacy and recognition programs.

>    1. Identifing, recruiting and engaging prospective projects.

Reviewer: ```suggestion
   1. Identifying, recruiting and engaging prospective projects.
```

   1. Voting members are responsible for making a final decision if consensus
      cannot be reached among members.
   1. Members of the CPC, or committees consisting of subsets of the CPC
      membership will be expected to meet on a regular basis to discuss topics
      such as new project acceptance into the mentorship program, project graduation
      from the mentorship program, Project issues and conflicts, opportunities
      for collaboration between Projects, opportunities for the Foundation in the
      greater JavaScript community, etc.

>   

Reviewer 3: I think we can make this paragraph simpler by splitting the responsibilities of voting members to the ones of regular members in two lists.

Author: lets do this in a follow up

     
## Section 6. Non-Responsibilities of the CPC

OpenJS Foundation Projects are self-governing entities. All policies and procedures for interacting with, contributing to and making decisions within a Project are defined, implemented, and documented by that Project’s maintainers and community during the mentorship process. Specific technical decisions within a Project will be the responsibility of the Project and establishment and implementation of the decision making process will be done within the guidelines as defined by CPC charter. Those Project decisions and responsibilities include but are not limited to:

* Setting release dates.
* Release quality standards.
* Technical direction.
* Project governance and process.
* GitHub repository hosting.
* Conduct guidelines.
* Maintaining the list of additional Collaborators.
* Development process and any coding standards.
* Mediating technical conflicts between Collaborators.

> Not withstanding the above, the Projects and the entire technical community will follow any processes

Reviewer: ```suggestion
Notwithstanding the above, the Projects and the entire technical community will follow any processes
```

as may be specified by the Board relating to the intake and license compliance review of contributions,
including the OpenJS Foundation IP Policy.

## Section 7. Elections

> Leadership roles in OpenJS Foundation will be peer elected

Reviewer: ```suggestion
Leadership roles in OpenJS Foundation will be peer-elected
```

the same person. The CPC shall hold annual elections to select a CPC
Chairperson and Director; there are no limits on the number of terms a
CPC Chairperson or Director may serve.

## Section 8. Board Representation

There are 3 Board seats earmarked for representation from the
OpenJS Foundation projects.  The first of these board seats is filled by
the CPC Director as outlined in Section 7 on [Elections][] and is 
always in place.  

The second board seat requires 2 Platinum board members to activate
and will be filled based on processes defined
by the CPC and approved by the Voting CPC members and then the Board. The
CPC members will review and either re-confirm or update the
processes once per year.

For the first year after the formation of the OpenJS Foundation, the
processes for filling this seat will be as follows:

* As a constellation member of the OpenJS Foundation, the Node.js project
  will elect one Board representative from either the Technical Steering Committee (TSC) or
  Community Committee (CommComm). The representative will be selected according to a process
  defined by the TSC and CommComm. 

As described in the [Elections][] section a multiple candidate method

> will be be used for the voting for the first and second board members.

Reviewer: ```suggestion
will be used for the voting for the first and second board members.
```
